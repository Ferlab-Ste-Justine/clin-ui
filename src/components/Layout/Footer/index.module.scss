@import "../../../style/themes/clin/colors";

.footer {
  padding: 0;

  .footerTop {
    background-color: $blue-9;
    width: 100%;
    height: 200px;
    align-items: center;

    .footerNav {
      width: 1200px;
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin: 0 auto;
      padding: 0 24px;

      .navTitle {
        font-weight: 500;
        letter-spacing: 0.25px;
        font-size: 20px;
        color: rgb(255, 255, 255);
        margin-bottom: 8px;
        display: block;
      }

      .logo {
        margin-top: -7px;
        height: 57px;
      }

      ul {
        list-style-type: none;
        margin: 0;
        padding: 0;
        font-size: 14px;
        letter-spacing: 0.25px;
        line-height: 20px;

        li {
          color: white;
          padding-bottom: 8px;

          &.externalLink {
            i {
              margin-left: 4px;
            }
          }

          i {
            margin-right: 4px;
          }

          a {
            color: #ffffff;
          }
        }
      }
    }
  }

  .footerBottom {
    background-color: #ffffff;
    width: 100%;
    height: 100px;

    .footerLogo {
      display: flex;
      justify-content: space-between;
      width: 1200px;
      margin: 0 auto;
      padding: 0 24px;

      nav {
        display: flex;
        align-items: center;
        width: 90%;

        ul {
          padding: 0;
          margin: 0;
          list-style-type: none;
          width: 100%;

          li {
            display: inline-block;
            font-size: 14px;
            letter-spacing: 0.25px;
            line-height: 20px;
            padding-right: 2%;

            a {
              color: rgba(35, 55, 80, 0.78);
            }

            a:hover {
              color: #233750;
            }
          }
        }
      }
    }
  }
}
