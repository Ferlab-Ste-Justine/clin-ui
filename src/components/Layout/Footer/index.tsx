/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Col, Layout, Row } from 'antd';
import intl from 'react-intl-universal';
import get from 'lodash/get';

import styles from './index.module.scss';

const ZEPLIN_URL = get(window, 'CLIN.zeplinUrl', process.env.REACT_APP_ZEPLIN_URL);
const FHIR_CONSOLE_URL = get(window, 'CLIN.fhirConsoleUrl', process.env.REACT_APP_FHIR_CONSOLE_URL);

const Footer = () => (
  <Layout.Footer id="footer" className={styles.footer}>
    <Row align="middle" justify="space-between">
      <Col>
        <nav>
          <ul>
            <li>
              <a href={ZEPLIN_URL} target="_blank" rel="noreferrer">{intl.get('footer.navigation.zepplin')}</a>
            </li>
            <li>
              <a href={FHIR_CONSOLE_URL} target="_blank" rel="noreferrer">{intl.get('footer.navigation.fhir')}</a>
            </li>
          </ul>
        </nav>
      </Col>
      <Col>
        <img alt="Saint-Justine" className="logo" src="/assets/logos/chujs-color.svg" />
      </Col>
    </Row>
  </Layout.Footer>
);

export default Footer;
