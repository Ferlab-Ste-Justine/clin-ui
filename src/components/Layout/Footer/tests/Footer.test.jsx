import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../index';

describe('Footer', () => {
  test('should be as the snapshot', () => {
    const component = renderer.create(
      <Footer />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});