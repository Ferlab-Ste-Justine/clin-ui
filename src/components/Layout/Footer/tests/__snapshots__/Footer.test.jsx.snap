// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should be as the snapshot 1`] = `
<footer
  className="ant-layout-footer"
  id="footer"
>
  <div
    className="ant-row ant-row-space-between ant-row-middle"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <nav>
        <ul>
          <li>
            <a
              href="https://notebook.qa.cqgc.hsj.rtss.qc.ca"
              rel="noreferrer"
              target="_blank"
            >
              Zeppelin
            </a>
          </li>
          <li>
            <a
              href="https://fhir-console.qa.cqgc.hsj.rtss.qc.ca/home"
              rel="noreferrer"
              target="_blank"
            >
              Fhir
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="ant-col"
      style={Object {}}
    >
      <img
        alt="Saint-Justine"
        className="logo"
        src="/assets/logos/chujs-color.svg"
      />
    </div>
  </div>
</footer>
`;
